src/main/java/mate/academy/spring/dao/UserDaoImpl.java

```diff
@@ -0,0 +1,50 @@
+package mate.academy.spring.dao;
+
+import java.util.List;
+import mate.academy.spring.model.User;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    public UserDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public void add(User user) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't add user " + user + " to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<User> getAll() {
+        try (Session session = sessionFactory.openSession()) {
+            Query<User> getAllUsers = session.createQuery("from User", User.class);
+            return getAllUsers.getResultList();
```

Comment:
Никита зказал бы переименовать так:
```suggestion
            Query<User> query = session.createQuery("from User", User.class);
            return getAllUsers.getResultList();
```